spec/execution_tests.lua

```diff
@@ -1749,6 +1749,71 @@ function execution_tests.run(compile_file, backend, _ENV, only_compile)
         end)
     end)
 
+    describe("For-in loops", function()
+        compile([[
+            local function iter(state: any, current: any): (any, any)
+                if current as integer == #(state as {any}) then
+                    return nil, nil
+                end
+
+                return current as integer + 1, (state as {any})[current as integer + 1]
+            end
+
+            typealias iterfn = (any, any) -> (any, any)
+            local function xpairs(xs: {any}): (iterfn, any, any)
+                return iter, xs, 0
+            end
+
+
+            export function double(xs: {integer}): {integer}
+                local out: {integer} = {}
+                for i, x in xpairs(xs) do
+                    out[#out + 1] = x as integer * 2
+                end
+
+                return out
+            end
+
+            export function abs_list(xs: {float})
+                for i, x in xpairs(xs) do
+                    if (x as float) < 0.0 then
+                        xs[i] = -(x as float)
+                    end
+                end
+            end
+        ]])
+
+        it("", function()
+            run_test([[
+                local xs = test.double({1, 2})
+                assert(xs[1] == 2 and xs[2] == 4)
+
+                xs = {0.0, 2.1, -1.0, -4.5}
+                test.abs_list(xs)
+                assert(xs[1] == 0.0 and xs[2] == 2.1 and xs[3] == 1.0 and xs[4] == 4.5)
+            ]])
+        end)
```

Comment:
Although we only have one `compile` block for everything, we prefer having a separate  `it` block for each test case. That way, if a test case fails it is easier to know which one it was.

What is the difference between the test for `double` and the test for `abs_list`? If there is an important difference, that should be reflected in the name of the test that we pass to the `it` function. If there isn't an important difference, then I think we can get rid of the `abs_list` test and just keep the `double` one.